investos/portfolio/risk_model/factor_risk.py

```diff
@@ -1,23 +1,351 @@
+            )
+
+            y, X = (
+                df_current.loc[:, self._fos_return_uuid],
+                df_current.loc[
+                    :, (self._fos_risk_factor_uuids + self._fos_industry_factors)
+                ],
+            )
+            X.loc[:, "const"] = (
+                1  # need the intercept column, will convert this to country factor later
+            )
+            model = sm.OLS(y, X)
+            results = model.fit()
+            # store coefficients, intercept, feature_names, r2, and t-values into self._factor_returns based on the date
+            self._factor_returns[date] = {
+                "coefficients": results.params,
+                "intercept": results.params[
+                    "const"
+                ],  # the intercept needs the X col of all 1
+                "feature_names": self._fos_risk_factor_uuids
+                + self._fos_industry_factors,
+                "r2": results.rsquared_adj,
+                "t-values": results.tvalues,
+            }
+
+        self._generate_factor_returns()
+        self._generate_idiosyncratic_returns()
+        print("Done regression")
+        return
+
+    def _generate_idiosyncratic_returns(self):
+        """
+        Generate idio returns
+        """
+        print("\n generating idiosyncratic returns...")
+        # Merge the DataFrames on 'date'
+        self._df_idio = pd.merge(
+            self._df_loadings,
+            self._df_factor_returns.reset_index(),
+            on="datetime",
+            suffixes=("", "_factor_returns"),
+        )
+
+        # Multiplying matching columns
+        for col in self._fos_risk_factor_uuids + self._fos_industry_factors:
+            self._df_idio[f"calc_f_r_{col}"] = (
+                self._df_idio[col] * self._df_idio[f"returns_{col}"]
+            )
+
+        # Dropping the extra columns
+        self._df_idio = (
+            self._df_idio.drop(columns=self._fos_risk_factor_uuids)
+            .drop(columns=self._fos_industry_factors)
+            .drop(columns=self.cols_with_return)
+        )  # drop factor exposure columns
+        self._df_idio["factor_return_1d"] = self._df_idio[
+            [
+                f"calc_f_r_{col}"
+                for col in (self._fos_risk_factor_uuids + self._fos_industry_factors)
+            ]
+        ].sum(axis=1)
+        self._df_idio["factor_return_1d_error"] = (
+            self._df_idio["factor_return_1d"] - self._df_idio["return_1d"]
+        )
+        self._df_idio = self._df_idio[
+            [
+                "datetime",
+                "id",
+                "return_1d",
+                "factor_return_1d",
+                "factor_return_1d_error",
+            ]
+        ]
+        self._df_idio_returns = self._df_idio[
+            ["datetime", "id", "factor_return_1d_error"]
+        ].rename({"factor_return_1d_error": "idio_return"})
+        return
+
+    def _generate_factor_returns(self):
```

Comment:
What's the purpose of this method?

Reply:
Is it for reporting / documenting statistical significance of factors? Doesn't look required for generating risk model. If that's the case, we shouldn't be running it unless the user asks

Reply:
Also let's name it better / clean up the logic

Reply:
I see you're using this both for reporting and for setting `self._df_factor_summary[self.cols_with_return]`, which you are using to calculate the covariance matrix

Let's split the reporting functionality (t stats) and setting self._df_factor_summary[self.cols_with_return]. Also, _df_factor_summary is not a good name for a factor return df. Please rename